internal/dnfjson/dnfjson.go

```diff
@@ -35,13 +35,13 @@ func NewSolver(modulePlatformID string, arch string, cacheDir string) *Solver {
 }
 
 // Depsolve the given packages with explicit excludes using the solver configuration and provided repos
-func (s *Solver) Depsolve(includes []string, excludes []string, repos []RepoConfig) (Results, error) {
+func (s *Solver) Depsolve(pkgSet rpmmd.PackageSet, repos []RepoConfig) (Results, error) {
```

Comment:
I think I need to understand the intended relationship between the new `dnfjson` package and the `rpmmd` package. Will both stay around? If so: the comment I made previously also apply to `rpmmd`, we should distinguish between public structs and structs used for serialisation.

Reply:
My current idea is that `rpmmd` should provide the types to represent rpm packages and functions for managing those types.  `dnfjson` should handle the interface to `dnf-json` and any types that needs.  Conversions might need to happen between the two types, or we can redesign either to match when possible.

I'd like to model these two packages and their dependencies after the relationship between rpm and dnf.  I think that would make working with them easier to understand.  Which would probably mean that `rpmmd` shouldn't know (or care) about repositories and it shouldn't handle resolving dependencies.  Maybe this makes `rpmmd` just a simple data type package (and maybe that's redundant?).

Looking at [the current `rpmmd` types and methods](https://pkg.go.dev/github.com/osbuild/osbuild-composer@v0.0.0-20220326083336-68639b4bf9d6/internal/rpmmd), I'd like to keep package-related stuff (`Package`, `PackageList`, and their methods) in `rpmmd` and move repository-related stuff to `dnfjson`.